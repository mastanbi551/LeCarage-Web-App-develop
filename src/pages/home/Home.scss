@import "../../styles/style.scss";

.home_page p {
  width: 641px;
  height: 58px;
  // left: 399px;
  // top: 583px;

  font-family: $montserratFont;
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 29px;
  display: flex;
  align-items: center;
  text-align: center;
  text-transform: uppercase;
  margin: auto;

  color: $primaryWhite;
}
.footer_text {
  display: inline-block;
  bottom: 0px;
  position: absolute;
  letter-spacing: 0px;
  text-align: center;
  transform: translate(-50%, -50%);
  width: 204px;
  font-family: $montserratFont;
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 29px;
  text-transform: uppercase;
  color: white;
}
