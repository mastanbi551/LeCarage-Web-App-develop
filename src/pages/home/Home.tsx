import React, {useEffect} from 'react';
import './Home.scss';
import {ConstantStrings, ImagesPath} from '../../utils/constants';
import { useNavigate } from 'react-router-dom';
import { BackgroundImageContainer, LeCarageLogo } from '../../components';

export const Home = ()=>{
    const navigate = useNavigate();
    useEffect(() => {
        const timer = setTimeout(() => {
           return navigate('/login');
          }, 3000);
          return () => clearTimeout(timer);
    });
    return(
        <BackgroundImageContainer imageSource={ImagesPath.splashimg} >
            <>
                <div className='home_page'>
                    <LeCarageLogo source={ImagesPath.lecaragelogo} altText='lecarage_logo' name="home_page_logo" />
                    <p>{ConstantStrings.splash_message}</p>
                
                </div>
                <div className='footer_text'>
                    <p>{ConstantStrings.lecarage_website}</p>
                </div>
            </>
           
        </BackgroundImageContainer>

        
    );
};