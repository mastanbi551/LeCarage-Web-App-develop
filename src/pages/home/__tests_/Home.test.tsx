import React from 'react';
import { render } from '@testing-library/react';
import { Home } from '../Home';
import { MemoryRouter } from 'react-router-dom';

describe('Home component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
