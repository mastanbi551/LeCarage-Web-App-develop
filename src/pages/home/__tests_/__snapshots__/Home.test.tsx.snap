// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="background-image-container"
    style="background-image: url(splash-image.png); background-size: cover; background-repeat: no-repeat; background-position: center; height: 100vh; overflow: hidden;"
  >
    <div
      class="home_page"
    >
      <img
        alt="lecarage_logo"
        class="home_page_logo"
        src="lecarageLogo.png"
      />
      <p>
        LUXURY CAR RESIDENCE CAR CLUB & PETROL HEADS COMMUNITY
      </p>
    </div>
    <div
      class="footer_text"
    >
      <p>
        LeCarage.com
      </p>
    </div>
  </div>
</DocumentFragment>
`;
